pset6/credit.py

```diff
@@ -0,0 +1,36 @@
+#http://docs.cs50.net/problems/credit/credit.html
+def check_credit(card_number):
+
+    card_length = len(card_number)
+    sum = 0
+
+    for i in range(card_length - 2, -1, -2):
```

Comment:
This would be better expressed in higher level terms. It's really hard to figure out what code does because it's just jumping around in the string. If you want the odd digits, grab the odd digits. If you want the even digits, grab the even digits, etc.

```
def add_digits(num):
  """Return the sum of the individual digits of num."""
  pass

# Grab odd digits going backwards starting from second to last
odds_backwards = (int(i) for i in card_number[-2::-2])

# Multiply each by 2 and add up the digits of each
odds_added = tuple(add_digits(i*2) for i in odds_backwards)

# Grab even digits going backwards starting from end (these are the remaining digits)
evens_backwards = tuple(int(i) for i in card_number[-1::-2])

# Grand total should be congruent to 10
check =  sum(odds_added + evens_backwards) % 10 == 0

return check
```

Kind of winging it here so there might be bugs, but it's much easier now to verify that each step works correctly and see what the original algorithm was doing.